fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -406,6 +411,189 @@ private boolean processDropColumn(DropColumnClause alterClause, OlapTable olapTa
+            throw new DdlException("Column[" + modifyColumnName + "] type is invalid, add field only support Struct");
+        }
+
+        List<String> nestedFieldName = alterClause.getNestedParentFieldNames();
+        Type modifyFieldType = getModifiedType(modifyColumn, nestedFieldName);
+        String modifyFieldName = null;
+        if (nestedFieldName == null || nestedFieldName.isEmpty()) {
+            modifyFieldName = modifyColumn.getName();
+        } else {
+            modifyFieldName = nestedFieldName.get(nestedFieldName.size() - 1);
+        }
+        if (!modifyFieldType.isStructType()) {
+            throw new DdlException("Field " + modifyFieldName + " is invalid, add field only support for Struct");
+        }
+
+        StructType oriFieldType = ((StructType) modifyFieldType);
+        String fieldName = alterClause.getFieldName();
+        Type fieldType = alterClause.getType();
+        StructField addField = new StructField(fieldName, id, fieldType, null);
+        ColumnPosition fieldPos = alterClause.getFieldPos();
+        ArrayList<StructField> oriFields = oriFieldType.getFields();
+        int posIndex = -1;
+        boolean hasPos = (fieldPos != null); 
+        if (fieldPos != null) {
+            if (fieldPos.isFirst()) {
+                posIndex = 0;
+            } else {
+                if (oriFieldType.getField(fieldPos.getLastCol()) == null) {
+                    throw new DdlException("Field[" + fieldPos.getLastCol() + "] is not exist in Field[" + modifyFieldName + "]");
+                }
+                posIndex = oriFieldType.getFieldPos(fieldPos.getLastCol()) + 1;
+            }
+        } else {
+            posIndex = oriFields.size();
+        }
+
+        ArrayList<StructField> fields = new ArrayList<>();
+        for (StructField field : oriFieldType.getFields()) {
+            fields.add(field);
+        }
+        fields.add(posIndex, addField);
+        oriFieldType.updateFields(fields);
+        for (Map.Entry<Long, LinkedList<Column>> entry : indexSchemaMap.entrySet()) {
+            List<Column> modIndexSchema = entry.getValue();
+            Optional<Column> oneCol = modIndexSchema.stream().filter(c -> c.nameEquals(modifyColumnName, true)).findFirst();
+            if (!oneCol.isPresent()) {
+                continue;
+            } else {
+                int idx = modIndexSchema.indexOf(oneCol.get());
+                modIndexSchema.set(idx, modifyColumn);
+            }
+        }
+
+        return fastSchemaEvolution;
+    }
+
+    /**
+     * @param alterClause
+     * @param olapTable
+     * @param indexSchemaMap
+     * @param indexes
+     * @return true: can light schema change, false: cannot
+     * @throws DdlException
+     */
+    private boolean processDropField(DropFieldClause alterClause, OlapTable olapTable,
+                                     Map<Long, LinkedList<Column>> indexSchemaMap, List<Index> indexes) throws DdlException {
```

Comment:
it seems that indexes is not used in this function.

Reply:
> it seems that indexes is not used in this function.

updated.